frontend/components/LocaleMenu.tsx

```diff
@@ -0,0 +1,202 @@
+import { Box, chakra, Checkbox, FormLabel } from "@chakra-ui/react";
+import Link from "next/link";
+import React, { PropsWithChildren, useState } from "react";
+import { MenuContext, useMenuContext } from "./MenuContext";
+
+export function LocaleMenu(props: PropsWithChildren<unknown>) {
+  const [isOpen, setIsOpen] = useState(false);
+  function toggle() {
+    setIsOpen((isOpen) => !isOpen);
+  }
+
+  return (
+    <MenuContext.Provider value={{ isOpen, toggle }}>
+      <React.Fragment>
+        <GooeySVGDefs />
+        <Box filter="url(#shadowed-goo1)" overflow="visible">
+          {props.children}
+        </Box>
+      </React.Fragment>
+    </MenuContext.Provider>
+  );
+}
+
+export function LocaleMenuButton(props: PropsWithChildren<unknown>) {
+  const value = useMenuContext();
+  return (
+    <Box className="checkbox e label">
+      <Checkbox
+        userSelect="none"
+        type="checkbox"
+        name="gooey-menu-open"
+        id="gooey-menu-open"
+        display="none"
+        onChange={value.toggle}
+      />
+      <FormLabel
+        userSelect="none"
+        fontWeight="light"
+        fontFamily="Roboto Mono"
+        fontSize="xs"
+        pos="relative"
+        backgroundColor="dark"
+        borderRadius="100%"
+        display="block"
+        zIndex="1"
+        left="0.5"
+        w="12"
+        h="12"
+        color="white"
+        textAlign="center"
+        lineHeight="48px"
+        htmlFor="gooey-menu-open"
+        transitionDuration="400ms"
+        transform={
+          value?.isOpen === true
```

Comment:
Why is `value` optional here? it shouldn't since you're already checking for nullability inside the `useMenuContext` hook

Reply:
Also, boolean variables like "isOpen" do not require explicit `=== true`